### PR TITLE
General-purpose tests of unsupported properties

### PR DESCRIPTION
**Pull request by OldSneerJaw**
Added several test cases to verify that unsupported properties are rejected when specified at the root level of a document or when nested inside an object.

### CODE REVIEW

File: test/general-spec.js [hunk 1]
```diff
@@ -129,6 +129,68 @@ describe('Functionality that is common to all documents:', function() {
     });
   });
 
+  describe('validator of supported properties', function() {
```
Comment (OldSneerJaw) [inline test/general-spec.js]: The specs for the sample document definitions indirectly test these behaviours as well, but this provides a more systematic means of verifying.

**Comment by ziemek** [comment]
Looks good! Must merge master.

**Comment by OldSneerJaw** [comment]
Merged. Ready for review again.

**Comment by ziemek** [comment]
👍

**Pull request merged**